destination.go

```diff
@@ -89,12 +89,14 @@ type destinationPluginAdapter struct {
 func (a *destinationPluginAdapter) Configure(ctx context.Context, req cpluginv1.DestinationConfigureRequest) (cpluginv1.DestinationConfigureResponse, error) {
 	ctx = DestinationWithBatch{}.setBatchEnabled(ctx, false)
 
-	var multiErr error
+	v := validator(a.impl.Parameters())
+	// init config and apply default values
+	updatedCfg, multiErr := v.initConfig(req.Config)
```

Comment:
Now that `initConfig` is meant to be called outside of the validator we should probably make it public (i.e. rename to `InitConfig`).

Reply:
should I take it out the validator type too? so just a static method? or keep it

Reply:
I'd leave it on the validator rather than have a static function. The function needs the map of parameters and the validator already has access to them. If we feel like this should not be part of the validator I'd rather rename `validator` and give it a more generic name like `parameterUtil`. I personally can live with `validator` initializing the config though 😉

Reply:
done